logger.js

```diff
@@ -0,0 +1,247 @@
+// Logger Utility with Linter Issues
+
+// Logger Levels
+const levels = {
+    info: "INFO",
+    warn: "WARN",
+    error: "ERROR",
+    debug: "DEBUG"
+}
+
+// Unused global variables
+var currentLogLevel = "info"
+let isLoggerEnabled = true
+
+// Function to get current timestamp
+function getTimeStamp(){
+const date = new Date();
+return date.getFullYear() + "-" + (date.getMonth() + 1) + "-" + date.getDate() +
+  " " + date.getHours() + ":" + date.getMinutes() + ":" + date.getSeconds()
+}
+
+// Base Logger Function
+function log(level, message, details) {
+  if(!isLoggerEnabled) return;
+  let timeStamp = getTimeStamp();
+  let logEntry = `[${timeStamp}] [${level}] - ${message}`
+  if (details) { logEntry += " - Details: " + JSON.stringify(details) }
+  console.log(logEntry)
+  saveLogToFile(logEntry) // Calls a function to save to file
+}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding parameter validation in log function</b></div>
<div id="fix">

The `log` function could benefit from error handling for invalid log levels and message validation. Consider adding checks for `level` and `message` parameters.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -23,6 +23,10 @@
 function log(level, message, details) {
 +  if (!Object.values(levels).includes(level)) {
 +    console.error(`Invalid log level: ${level}`);
 +    return;
 +  }
 +  if (!message || typeof message !== 'string') {
 +    console.error('Invalid message format');
 +    return;
 +  }
   if(!isLoggerEnabled) return;
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/BitoPythonTest/pull/20#issuecomment-2440426565>#7b8f80</a></i></small>
</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged